### ISSUE TITLE
predictive modeling with databot

### ISSUE THREAD

**Issue by simonpcouch** [jcheng5/databot#1]
Assorted notes from using the databot for predictive modeling with tidymodels:

**Prompting / the model**

[Here's a draft prompt](https://gist.github.com/simonpcouch/8c8886dfac6d102f8b96eae5ab431cde) I put together while experimenting.

* It's quite good at feature engineering and model diagnostics (which I guess makes sense since that's language independent?)
* In general, most of the issues I see are a matter of poise/approach:
	* It's too eager to evaluate on the test set.
	* It generates preprocessing, training, resampling and evaluation code on the first response unless explicitly instructed to stop after each step.
* Claude tends to write recipes, parsnip, and rsample code pretty fluently, but once it's time to evaluate performance, it starts to improvise functions rather than relying on tidymodels' defaults and/or yardstick functions.
* It _really_ wants to use parallelism, and does so using the deprecated foreach framework. I've yet to figure out how to prompt it to prefer future over foreach without it just always using parallelism, even when the user doesn't ask for it.
* When fitting models, I've never seen it recommend anything other than linear models, random forests, or boosted trees with XGBoost. A reasonable "top three" though, I guess.
* Including the information on how to build predictive models in the prompt tends to result in the model suggesting that one build models even when they're just doing EDA. Not an issue for someone purposefully including `llms.txt` with that information, but maybe there ought to be some sort of "Knowledge packs" functionality where users select which bits of prompting are included in the context on startup—those could be arguments to `databot::chat()` initially and possibly a dropdown in a more productized UI.
* In general, I don't see the model being quite as inquisitive/imaginative when modeling as I do when doing EDA. This could likely be a result of the prompts recommending such a structured approach, but I don't see the model adapting its recommended approaches based on the input data.

**Interface**

In general, this comes down to modeling code often taking a good bit longer to run than EDA code. 

* In moments where code is taking a while to evaluate, I'm 1) a bit unsure if anything's actually happening and 2) a bit bummed knowing that all of that computation will need to be redone when generating the initial draft of the report _and_ once the users makes edits. Almost makes me wonder if the model ought not to try and render the report on its first go and automatically set up caching as it does so.
* A bit painful to see when the model generates code that I immediately see will take a long time to run but have to wait for the code to finish running before I can do anything about it. (This would be resolved if databot supported interruption.)
* Another side effect of this code taking longer to run is that I really miss the tidymodels progress bar functionality in the console. Maybe this is just because I've spent a lot of time on that functionality, but seeing that logging would make the process feel much more interactive and also assure me that some progress is being made when tuning models.

**Comment by jcheng5** [comment]
These are incredibly interesting observations! It's kind of exciting to see how different the needs are on the modeling side, it makes me more confident that we have a real contribution to make here.
